core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -167,8 +171,8 @@ export class Question {
   }
 
   toBackendDict(isNewQuestion: boolean): QuestionBackendDict {
-    var questionBackendDict = {
-      id: null,
+    var questionBackendDict: QuestionBackendDict = {
+      id: 'id',
```

Comment:
Why this change? You should either get rid of this function altogether or provide proper ID.

Reply:
> Why this change? You should either get rid of this function altogether or provide proper ID.

So regarding this, I would like to know the basic flow before doing anything like, at what stage exactly does the id actually get's set and how is it generated, is it that it has to take null value here and checks if it's null and then generate a new ID for it, if so maybe we can directly assign a new id here only.

Reply:
To do this, we would have to change what question editor expects in backend in new question to have id `None` to id `''` empty string like this or it will give an error when creating a new question
![image](https://user-images.githubusercontent.com/42764756/143386639-b932de0b-2c7d-4289-86ef-35942b3fc5c1.png)

Reply:
@vojtechjelinek Will make this update once you approve. Thanks!

Reply:
No, an empty string is not a good replacement for `null`, we should either get rid of the `null` or keep it and document why is it needed.

Reply:
> 

Sure then, will keep the null and document it. Thanks.

Reply:
@vojtechjelinek Made appropriate changes. Please take a look.